autoload/clang_format.vim

```diff
@@ -291,6 +311,92 @@ endfunction
 function! clang_format#disable_auto_format() abort
     let g:clang_format#auto_format = 0
 endfunction
+
+" s:strip: helper function to strip a string
+function! s:strip(string)
+    return substitute(a:string, '^\s*\(.\{-}\)\s*\r\=\n\=$', '\1', '')
+endfunction
+
+" clang_format#get_git_diff
+" a:file must be an absolute path to the file to be processed
+" this function compares the current buffer content against the
+" git index content of the file.
+" this function returns a list of pair of ranges if the file is tracked
+" and has changes, an empty list otherwise
+function! clang_format#get_git_diff(cur_file)
+    let file_path = isdirectory(a:cur_file) ? a:cur_file :
+                \ fnamemodify(a:cur_file, ":h")
+    let top_dir=s:strip(system(
+                \ g:clang_format#git." -C '".file_path."' ".
```

Comment:
```suggestion
                \ g:clang_format#git." -C ".shellescape(file_path)." ".
```

Simply using single quotes does not work on some shell. Can you use `shellescape` instead?